source/frontend/src/components/maps/MapSearch.tsx

```diff
@@ -0,0 +1,88 @@
+import { dequal } from 'dequal';
+import React, { useEffect, useState } from 'react';
+import { Container } from 'react-bootstrap';
+import styled from 'styled-components';
+
+import { useMapStateMachine } from '@/components/common/mapFSM/MapStateMachineContext';
+import { PropertyFilter } from '@/features/properties/filter';
+import {
+  defaultPropertyFilter,
+  IPropertyFilter,
+} from '@/features/properties/filter/IPropertyFilter';
+
+export type MapSearchProps = {};
+
+/**
+ * Creates a Leaflet map and by default includes a number of preconfigured layers.
+ * @param param0
+ */
+const MapSearch: React.FC<React.PropsWithChildren<MapSearchProps>> = () => {
+  const { mapFilter: mapMachineMapFilter, setMapFilter: mapMachineSetMapFilter } =
+    useMapStateMachine();
+
+  const [mapFilter, setMapFilter] = useState<IPropertyFilter | null>(null);
+
+  useEffect(() => {
+    if (mapFilter !== null && !dequal(mapFilter, mapMachineMapFilter)) {
```

Comment:
can you help me understand why this shouldn't go in the handle function below?

Reply:
It is a problem of initialization. This component gets initialized whenever the application starts and when the sidebar closes. On the second case the map fsm knows the filter already, while on the first case it does not and expects the url parser on the PropertyFilter to parse the url and update the state.
In summary, the handleMapFilterChange gets called on initialization always, but is not always necessary to set the map filter on the map fsm